### PR TITLE
[New Rule] Environment Variable Modification using Launchctl

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/864

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/defense_evasion_modify_environment_launchctl.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+creation_date = "2021/01/14"
+maturity = "production"
+updated_date = "2021/01/14"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies modification of an environment variable using launchctl built-in command. Adversaries may execute their own
+malicious payloads by hijacking certain environment variables to load arbitrary libraries or bypass certain
+restrictions.
```
Comment (bm11100) [inline rules/macos/defense_evasion_modify_environment_launchctl.toml:11]:
```suggestion
Identifies modifications to an environment variable using the launchctl built-in command. Adversaries may execute their own
malicious payloads by hijacking certain environment variables to load arbitrary libraries or bypass certain
restrictions.
```

File: rules/macos/defense_evasion_modify_environment_launchctl.toml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Environment Variable Modification using Launchctl"
```
Comment (bm11100) [inline rules/macos/defense_evasion_modify_environment_launchctl.toml:17]:
```suggestion
name = "Modification of Environment Variable via Launchctl"
```

Comment (bm11100) [reply rules/macos/defense_evasion_modify_environment_launchctl.toml:17]: for consistency with other rule names

File: rules/macos/defense_evasion_modify_environment_launchctl.toml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+risk_score = 47
+rule_id = "7453e19e-3dbf-4e4e-9ae0-33d6c6ed15e1"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Defense Evasion"]
+type = "query"
+
+query = '''
+event.category:process and event.type:(start or process_started) and
+  process.name:launchctl and
+  process.args:(setenv and not (JAVA*_HOME or RUNTIME_JAVA_HOME or DBUS_LAUNCHD_SESSION_BUS_SOCKET or ANT_HOME))
```
Comment (bm11100) [inline rules/macos/defense_evasion_modify_environment_launchctl.toml:30]: could benefit from case insensitivity but will leave up to you

**Review by bm11100 — APPROVED** [review]

File: rules/macos/defense_evasion_modify_environment_launchctl.toml (hunk 3 above)
Comment (Samirbous) [reply rules/macos/defense_evasion_modify_environment_launchctl.toml:30]: not necessary: most process args are sensitive and cmd will fail if insensitive.

File: rules/macos/defense_evasion_modify_environment_launchctl.toml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+creation_date = "2021/01/14"
+maturity = "production"
+updated_date = "2021/01/14"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies modifications to an environment variable using the launchctl built-in command. Adversaries may execute their own
+malicious payloads by hijacking certain environment variables to load arbitrary libraries or bypass certain
+restrictions.
```
Comment (peasead) [inline rules/macos/defense_evasion_modify_environment_launchctl.toml:11]:
```suggestion
Identifies modifications to an environment variable using the built-in launchctl command. Adversaries may execute their own
malicious payloads by hijacking certain environment variables to load arbitrary libraries or bypass certain
restrictions.
```

File: rules/macos/defense_evasion_modify_environment_launchctl.toml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+risk_score = 47
+rule_id = "7453e19e-3dbf-4e4e-9ae0-33d6c6ed15e1"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Defense Evasion"]
+type = "query"
+
+query = '''
+event.category:process and event.type:(start or process_started) and
+  process.name:launchctl and
+  process.args:(setenv and not (JAVA*_HOME or RUNTIME_JAVA_HOME or DBUS_LAUNCHD_SESSION_BUS_SOCKET or ANT_HOME))
```
Comment (peasead) [inline rules/macos/defense_evasion_modify_environment_launchctl.toml:30]:
```suggestion
event.category:process and event.type:start and
  process.name:launchctl and
  process.args:(setenv and not (JAVA*_HOME or RUNTIME_JAVA_HOME or DBUS_LAUNCHD_SESSION_BUS_SOCKET or ANT_HOME))
```

**Review by peasead — APPROVED** [review]
LGTM

**Pull request merged**